warehouse/forklift/legacy.py

```diff
@@ -572,6 +572,14 @@ def _is_valid_dist_file(filename, filetype):
     return True
 
 
+def _is_duplicate_file(request, filename, sha256_hash):
```

Comment:
One other thing, it'd probably be a tad bit nicer to have this take a ``Session()`` object instead of a ``Request()`` object. That would mean that you just pass in ``request.db`` instead of ``request``, but it then means that (A) this function can be tested independently of a request and (B) if we ever want to re-use this function in a context we don't have a request, we can. It's also just a bit more hygienic since the only reason we need the ``Request()`` object is to access the ``Session()`` object hanging off it at ``request.db``.

Reply:
I felt bad passing around a `Request()` as it were, so this is the feedback I was looking for. Thanks again!